autogen/agentchat/assistant_agent.py

```diff
@@ -15,9 +18,12 @@ class AssistantAgent(ConversableAgent):
     This agent doesn't execute code by default, and expects the user to execute the code.
     """
 
-    DEFAULT_SYSTEM_MESSAGE = """You are a helpful AI assistant.
+    shell_command = "sh" if sys.platform != "win32" else powershell_command
```

Comment:
There is another issue with this. The code execution might be running inside a docker container which may not be Windows, even though the agent' itself is running in Windows. So using the agent's platform may not be correct in this case.

Reply:
Oh okay that makes sense then. I thought the agent will run on the same platform where the code execution takes place.

Reply:
I can add another condition which checks if the code execution is happening in a docker container or not though. That should fix this issue temporarily until I make the prompt organization module right?

Reply:
The code executor agent is often not the agent that writes the code. So you would need to set the platform string of the agent that writes the code based on the code executor of the agent that runs the code. It can get more complicated in a group chat when you have multiple code executor agents and code writing agents.

Reply:
Can you check the condition I added for docker now?

Reply:
> Collaborator

Ohh god okay. Should I even try doing that in this PR then?

Reply:
> The code executor agent is often not the agent that writes the code. So you would need to set the platform string of the agent that writes the code based on the code executor of the agent that runs the code. It can get more complicated in a group chat when you have multiple code executor agents and code writing agents.

Ohh god okay. SHould I even try doing that in this PR then?

Reply:
Could you give an example where code is being executed by some function other than the execute_code function?

Reply:
@ekzhu I tried to solve this, could you review my code again please?